src/resources/resolver.js

```diff
@@ -82,7 +83,27 @@ export function buildEndpoint(endpoint, params = {}, options = {}) {
     return endpoint;
   }
 
-  return new UriTemplate(endpoint).expand(params);
+  const usedParams = [];
+  const template = new UriTemplate(endpoint);
+  const templateUri = template.expand(key => {
+    usedParams.push(key);
+    return params[key];
+  });
```

Comment:
I don't really understand this part. Before, `template` was expanded with params.
Now, you're sending an arrow function as the `expand` input parameter, but I don't see how it will be called? I expect the arrow function should be called for every parameter in `params` object, but written like this, I think it will never be invoked..?
Correct me if I'm wrong.

Reply:
Docs: https://medialize.github.io/URI.js/uri-template.html
The endpoint has it's own params, for example, users/{userId}/cars , template expand will call arrow function with every key that is in endpoint, in this example, it is "userId". Logic is that we can figure out which keys are already populated by UriTemplate and unused params will be populated by us. We need to add unused params too so that we are compatible with redux-io v1, so that Shoutem builder can use version 3.